Ladybird/LocationEdit.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2023, Cameron Youell <cameronyouell@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "LocationEdit.h"
+#include "Tab.h"
+#include "Utilities.h"
+#include <AK/URL.h>
+#include <QCoreApplication>
+#include <QPalette>
+#include <QTextLayout>
+
+LocationEdit::LocationEdit(Tab* tab)
```

Comment:
ah, one last thing. I don't think that we actually need to know about Tab in this class do we? This class could just take a QWidget as its parent and pass that to the QLineEdit constructor, right? Then we could forget about Tab.h in the cpp and the forward decl in the header..

Reply:
you're correct! All Fixed now :)